### PR TITLE
[#10765] Add test coverage for TrieMaps with null values

### LINKED ISSUE #10765: Odd Undocumented TrieMap Behavior with null Values
Scala version: 2.12.4
java version: 1.8.0_161
OS:  ubuntu 16.04 LTS

The following is an issue I encountered with mapping a key to null in TrieMap (I have not confirmed that this behavior occurs with other Map implementations):

```scala
scala> /* let's start with a TrieMap with "a" mapped to null */
     | val trieMap = scala.collection.concurrent.TrieMap[String, String]("a" -> null)
trieMap: scala.collection.concurrent.TrieMap[String,String] = TrieMap(a -> null)

scala> /* looks good, let's retrieve the value associated with "a" */
     | trieMap("a")
java.util.NoSuchElementException
  at scala.collection.concurrent.TrieMap.apply(TrieMap.scala:837)
  ... 35 elided

scala> /* wait... what?  It just showed it was mapped to null */
     | trieMap.get("a")
res4: Option[String] = None

scala> /* really?  So this should return false */
     | trieMap contains "a"
res6: Boolean = false

scala> /* And so it does.  So how many entries are there? */
     | trieMap.size
res8: Int = 1

scala> /* So... to what is that one entry mapped? */
     | val iterator = trieMap.iterator
iterator: Iterator[(String, String)] = non-empty iterator

scala> /* there is an item there, isn't there? */
     | iterator.hasNext
res11: Boolean = true

scala> /* ok, so what is it? */
     | iterator.next
res13: (String, String) = (a,null)

scala> /* right back where we started... */
```
So is TrieMap supposed allow null values?  If so, let them be queried and retrieved just like all other mapped values.  If not, don't allow it as a value in the first place.  My preference is to allow null values, but either way, please document the behavior thoroughly.

### PR DESCRIPTION
**Pull request by joshlemer**
closes scala/bug#10765

### CODE REVIEW

**Review by julienrf — APPROVED** [review]

**Comment by SethTisue** [comment]
looks like our Jenkins disk space troubles affected this. rebasing should un-confuse Scabot

**Comment by joshlemer** [comment]
@SethTisue this is ready

**Comment by SethTisue** [comment]
thank you Josh!

**Comment by joshlemer** [comment]
No problem :-)

**Comment by Jasper-M** [comment]
You know you're testing `TreeMap` instead of `TrieMap`?

File: test/junit/scala/collection/concurrent/TrieMapTest.scala [hunk 1]
```diff
@@ -59,11 +62,22 @@ class TrieMapTest {
...
     // are different keys (collision)
-    Assert.assertEquals(Some("zero"), xs.get(8))
-    Assert.assertEquals(Some("four"), xs.get(4))
+    assertEquals(Some("zero"), xs.get(8))
+    assertEquals(Some("four"), xs.get(4))
+  }
+
+  @Test
+  def nullValues_t10765: Unit = {
+    val trieMap = TreeMap[String, String]("a" -> null)
```
Comment (Jasper-M) [inline test/junit/scala/collection/concurrent/TrieMapTest.scala:75]: You know you're testing `TreeMap` instead of `TrieMap`?

Comment (joshlemer) [reply test/junit/scala/collection/concurrent/TrieMapTest.scala:75]: Whoops, will fix it!

**Comment by joshlemer** [comment]
~@SethTisue this should be reopened~

**Comment by SethTisue** [comment]
@joshlemer can't reopen a merged PR. you'll just have to make a followup PR

**Comment by SethTisue** [comment]
@joshlemer does this need to be revisited?

**Comment by joshlemer** [comment]
@SethTisue nope! Read 'em and weep https://github.com/scala/scala/pull/7087/files#diff-b30cd07971c5a3d008732de5619e635cR85
